src/components/trip_view/cal/ToDoView.js

```diff
@@ -0,0 +1,79 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import { ToDoItem } from './ToDoItem';
+import { filterTripToDos } from '../../../redux/getters/getEvents';
+
+/**
+ * This react-big-calendar view displays all events that have a NULL
+ * start time. That is, it shows all unscheduled to-dos.
+ * It takes much of its structure from the AgendaView component.
+ */
+class ToDoList extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.renderToDo = this.renderToDo.bind(this);
+  }
+
+  render() {
+    let { events, localizer } = this.props;
+    let { messages } = localizer;
+
+    // Gets just events with no associated start time.
+    events = filterTripToDos(events);
+
+    // Sort by date (the + turns them into numbers that are easily sortable)
+    events.sort((a, b) => +b.priority - +a.priority);
+
+    // Figure out what to show for the to-do list
+    // (depends on if there are any events)
+    let agendaContent;
+    if (events.length !== 0) {
+      agendaContent = (
+        <div className="rbc-agenda-content" ref="content">
+          {events.map((event) => {
+            return this.renderToDo(event);
+          })}
+        </div>
+      );
+    } else {
+      agendaContent = (
+        <span className="rbc-agenda-empty">There are no unscheduled to-dos.</span>
+      );
+    }
+
+    return <div className="rbc-agenda-view">{agendaContent}</div>;
+  }
+
+  /**
+   * This renders a single to-do (event)
+   * @param event the event to render (with no dates)
+   * @returns {*} JSX for the event view
+   */
+  renderToDo(event) {
```

Comment:
Didn't know you could nest render-able object like this, neat.

Reply:
One of the most beautiful parts about react 😄